test/integration/container/multimaster_test.go

```diff
@@ -284,86 +378,101 @@ func TestMultimasterSetup(t *testing.T) {
-	node0IP = sanitizeIP(run(t, "docker", "inspect", node_os+"-multimaster-node0", "--format='{{.NetworkSettings.IPAddress}}'"))
-	fmt.Printf("node0IP: %s\n", node0IP)
-	node1IP = sanitizeIP(run(t, "docker", "inspect", node_os+"-multimaster-node1", "--format='{{.NetworkSettings.IPAddress}}'"))
-	node2IP = sanitizeIP(run(t, "docker", "inspect", node_os+"-multimaster-node2", "--format='{{.NetworkSettings.IPAddress}}'"))
-	node3IP = sanitizeIP(run(t, "docker", "inspect", node_os+"-multimaster-node3", "--format='{{.NetworkSettings.IPAddress}}'"))
-
-	dirName := tempDir(t)
-	clusterYamlContent := fmt.Sprintf(clusterYAML, registryIP, registryPort)
-	clusterYaml := saveToFile(t, dirName, "cluster.yaml", clusterYamlContent)
-	fmt.Printf("clusterYAML file: %s\ncontents: \n%+v\n", clusterYaml, clusterYamlContent)
-
-	machinesYamlContent := fmt.Sprintf(string(machinesYAML), node0IP, node1IP, node2IP, node3IP)
-	machinesYaml := saveToFile(t, dirName, "machines.yaml", machinesYamlContent)
-	fmt.Printf("machinesYAML file: %s\ncontents: \n%+v\n", machinesYaml, machinesYamlContent)
-
-	_ = saveToFile(t, dirName, "repo-config.yaml", fmt.Sprintf(repoConfigMap, yumRepoIP))
-	_ = saveToFile(t, dirName, "docker-config.yaml", fmt.Sprintf(dockerConfigMap, registryIP, registryPort))
-
-	run(t, "../../../cmd/wksctl/wksctl", "apply",
-		fmt.Sprintf("--cluster=%s", clusterYaml), fmt.Sprintf("--machines=%s", machinesYaml),
-		fmt.Sprintf("--config-directory=%s", dirName),
-		"--verbose",
-		fmt.Sprintf("--controller-image=docker.io/weaveworks/wksctl-controller:%s", tag))
-
-	out := run(t, "../../../cmd/wksctl/wksctl", "kubeconfig",
-		fmt.Sprintf("--cluster=%s", clusterYaml), fmt.Sprintf("--machines=%s", machinesYaml))
-
-	var nodeList corev1.NodeList
-	for {
-		jsonOut, stderr := doRun("kubectl", "get", "nodes", "-o", "json", fmt.Sprintf("--kubeconfig=%s", kubeconfig(out)))
-		if stderr != "" {
-			log.Warnf("error from kubectl; ignoring: %s", stderr)
-			continue
-		}
-		if err := json.Unmarshal([]byte(jsonOut), &nodeList); err != nil {
-			log.Warnf("Error deserialising output of kubectl get nodes: %s", err)
-		}
-		log.Infof("The cluster currently has %d node(s)", len(nodeList.Items))
-		if len(nodeList.Items) == 4 {
-			break
-		}
-		time.Sleep(30 * time.Second)
-	}
-
-	assert.Len(t, nodeList.Items, 4)
-	assert.Len(t, nodes.Masters(nodeList).Items, 3)
-	assert.Len(t, nodes.Workers(nodeList).Items, 1)
 
-	expectedKubeletArgs := []string{"alsologtostderr=true", "container-runtime=docker"}
-	expectedApiServerArgs := []string{"alsologtostderr=true", "audit-log-maxsize=10000"}
+	tests := []struct {
+		name string
+		path string
+		skip bool
+	}{
+		{
+			name: CENTOS,
+			path: "../../../examples/footloose/centos7/docker/multimaster.yaml",
+			skip: false,
+		},
+		{
+			name: UBUNTU,
+			path: "../../../examples/footloose/ubuntu1804/docker/multimaster.yaml",
+			skip: true,
```

Comment:
How do we select Ubuntu (or get both to run) now?

Reply:
It should still be possible to select the node_os you'd like to run.  There are two ways to go about it:

1. `NODE_OS=<ubuntu|centos> go test -v test/integration/container...`
2. `NODE_OS=<ubuntu|centos> make integration-tests-container`